client/packages/common/src/ui/components/navigation/Tabs/DetailTabs.tsx

```diff
@@ -29,9 +29,14 @@ export const DetailTabs: FC<DetailTabsProps> = ({
   tabs,
   requiresConfirmation = () => false,
 }) => {
+  const isValidTab = (tab?: string): tab is string =>
+    !!tab && tabs.some(({ value }) => value === tab);
   const { urlQuery, updateQuery } = useUrlQuery();
-  const [currentTab, setCurrentTab] = useState<string>(tabs[0]?.value ?? '');
   const t = useTranslation();
+  const currentUrlTab = urlQuery['tab'] as string | undefined;
```

Comment:
relying on the `useEffect` resulted in the first tab always loading. For stocktake detail view, trying to view the log was super slow because of this.